File: docs/site/samples/typescript/examples/guides_guard_route_single.ts
// Routing the guard text for a single render: the advisory goes to a system
// message, the body to a user message. The library never concatenates them.
// Standalone — run under `node --test`.

import assert from "node:assert/strict";
import { readFileSync } from "node:fs";
import { fileURLToPath } from "node:url";
import { test } from "node:test";
import { Prompt } from "prompting-press";
import { z } from "zod";

// The caller reads the definition; the library does no file I/O itself.
// Resolve the file next to this program (a real app uses its own path).
const defFile = (name: string) => fileURLToPath(new URL(name, import.meta.url));

const ask = Prompt.fromYaml(readFileSync(defFile("ask.yaml"), "utf8"));

const Ask = z.object({ topic: z.string().min(1) });

test("route the guard advisory to a system message beside the user body", () => {
	const result = ask.render(
		Ask,
		{ topic: "rivers" },
		{ guard: { enabled: true } },
	);

	const messages = [
		{ role: "system", content: result.guard },
		{ role: "user", content: result.text },
	];

	assert.equal(messages[0].role, "system");
	assert.ok(messages[0].content?.includes("<untrusted>"));
	assert.equal(messages[1].role, "user");
	assert.equal(
		messages[1].content,
		"Tell me about <untrusted>rivers</untrusted>.",
	);
});

File: docs/site/samples/python/examples/guides_guard_route_single.py
"""Routing the guard text for a single render: the advisory goes to a system
message, the body to a user message. The library never concatenates them.
Standalone — run directly or under pytest."""

from pathlib import Path

from prompting_press import Prompt, GuardConfig
from pydantic import BaseModel, Field

# The caller reads the definition; the library does no file I/O itself.
# Resolve the file next to this program (a real app uses its own path).
_HERE = Path(__file__).parent
ask = Prompt.from_yaml((_HERE / "ask.yaml").read_text())


class Ask(BaseModel):
    topic: str = Field(min_length=1)


def main() -> None:
    result = ask.render(Ask, data={"topic": "rivers"}, guard=GuardConfig(enabled=True))

    messages = [
        {"role": "system", "content": result.guard},
        {"role": "user", "content": result.text},
    ]

    assert messages[0]["role"] == "system"
    assert "<untrusted>" in messages[0]["content"]
    assert messages[1]["role"] == "user"
    assert messages[1]["content"] == "Tell me about <untrusted>rivers</untrusted>."


if __name__ == "__main__":
    main()
